docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,623 @@
+
+```graphql title="Shipping subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  weight: Int! @external
+  shippingCost: Int! @requires("weight")
+}
+```
+
+In the example above, the Shipping subgraph adds the `shippingCost` field to the `Product` entity. This field requires the value of the `weight` field, which _cannot_ be resolved by the Shipping subgraph. Therefore, the Shipping subgraph defines the `weight` field but also marks it as `@external`.
+
+### `@inaccessible`
+
+```graphql
+directive @inaccessible(from: String!) on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION
+```
+
+The `@inaccessible` directive indicates that a location in the subgraph schema should _not_ be included in the [API schema](./federated-types/overview/#api-schema) that clients use to execute operations against the graph router.
+
+This directive is helpful while adding a new [value type](./federated-types/sharing-types/) field across multiple subgraphs. By marking the new field as `@inaccessible` in subgraphs that _do_ define it, the supergraph avoids composition errors caused by other subgraphs _not_ defining it.
+
+Whenever you've rolled out a definition across all relevant subgraphs, you can remove `@inaccessible` directive to begin including it in the composed API schema.
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/link/v1.0")
+  @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@inaccessible"])
+
+interface Product {
+  id: ID!
+  sku: String
+  package: String
+  createdBy: User
+  hidden: String @inaccessible
+}
+```
+
+
+### `@key`
+
+```graphql
+directive @key(fields: FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
+```
+
+The `@key` directive indicates a combination of fields that can be used to uniquely identify a particular instance of a federated entity:
+
+```graphql
+type Product @key(fields: "upc") {
+  upc: ID!
+  name: String
+}
+```
+
+Multiple keys can be defined on a single object type:
+
+```graphql
+type Product @key(fields: "upc") @key(fields: "sku") {
+  upc: ID!
+  sku: ID!
+  name: String
+}
+```
+
+> **Note:** Repeated directives (such as `@key` in the example above) require support by your underlying GraphQL implementation. Some subgraph libraries support only a single `@key` per entity because of this limitation.
+
+### `@link`
+
+```graphql
+directive @link(
+  url: String,
+  as: String,
+  for: link__Purpose,
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive links definitions within the document to external schemas.
+
+External schemas are identified by their `url`, which optionally ends with a name and version with the following format: `{NAME}`/v`{MAJOR}`.`{MINOR}`
+
+The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).
```

Comment:
Is this outdated?

Reply:
maybe, the way link is declared in that spec is slightly different that how it is declared in federation:

```
directive @link(
  url: String!,  # url is non-nullable here
  as: String,
  import: [Import],
  for: Purpose
) repeatable on SCHEMA
```